frontend/lib/moratorium-banner.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react'
+
+
+type Props = {}
+type State = {
+    isHidden: boolean
+  }
+  
+
+class MoratoriumBanner extends Component<Props,State> {
+    constructor(Props: Props) {
+      super(Props);
+  
+      this.state = {
+        isHidden: false
+      }
+  
+    }
+  
+    closeBanner = () => this.setState({isHidden: true});
```

Comment:
Er, also, given the thing's propensity to potentially re-pop-up even after a user has dismissed it, I'm wondering if, in addition to only putting it on the few pages you suggested, we _also_ remove the close button: my hypothesis is that an un-closeable alert can potentially be less annoying than a recalcitrant alert that keeps showing up over and over.

That said, yet _another_ option is to store the state of the alert's display in the browser's `localStorage`.  On the vast majority of browsers this would mean that shoo-ing away the alert once would indeed shoo it away forever, much the way some sites have that annoying "IS IT OK IF WE USE COOKIES" alert that only shows up once.

Anyhow, if you do end up keeping the close button, I recommend additionally wrapping it in a `<SimpleProgressiveEnhancement>` component--this will ensure it's not shown to users in compatibility mode, which means they won't see a close box that is completely non-functional.

(Also if you keep the close button, I think this would be a great opportunity for you to use [React hooks](https://reactjs.org/docs/hooks-intro.html) instead of making a class-based component, but given all the other priorities on your plate I understand if you pass on that!)

Reply:
Thank you so much for all this thorough feedback. Always amazing to see how much you think about all the different kinds of users and user scenarios that may come up in our products! 

In terms of removing the "X" button, Tahnee was very keen on having one as for our mobile users, it was important to be able to collapse the banner as it tended to get very expansive on smaller devices. That being said, I think it totally makes sense to get rid of it on all pages and only keep it on the 5 pages we discussed at product check-in. When I noticed that the component's state stayed consistent throughout the lifetime of the web page, I think I got a little too excited and thought we could just keep it as is to catch any case where a user was visiting the Tenant Platform from an external page. But, given the nuisances you pointed out, it definitely seems like a good move to limit the presence of the banner except for where we really need it.

Will also work on removing the X button for users in compatibility mode and implementing it as a `<button>` for screen readers.

Reply:
Awesome, that sounds great to me! 🎉